### DISCUSSION TITLE
Accessing controls inside a ContentControl (RE: title bar customisation)

### DISCUSSION THREAD

**Discussion by beeradmoore** [microsoft/WindowsAppSDK#2111] [Q&A] [answered]
I am wanting to create a custom title bar and even though the [new docs](https://docs.microsoft.com/en-us/windows/apps/develop/title-bar?tabs=wasdk) go into a lot of detail, specifically [Color and transparency in caption buttons](https://docs.microsoft.com/en-us/windows/apps/develop/title-bar?tabs=winui3#color-and-transparency-in-caption-buttons), but there are a few problems with it still, specifically with changing themes. I will go repro and report those bugs separately another time, but at the moment I have title bar exhaustion and I just want my app updated and published ASAP

I was able to overcome this in my test app by using a [FakeMinMaxCloseButtons](https://github.com/beeradmoore/WindowsAppSDK_NavColorTest/blob/main/NavTest/NavTest/UserControls/FakeMinMaxCloseButtons.xaml) class. The idea being if Win11 (eg. `IsCustomizationSupported`) we use defaults, if anything else we do a full replace of the titlebar, window caption buttons included.

For the fake window caption buttons I grab the grid from within `~\.nuget\packages\microsoft.windowsappsdk\1.0.0\lib\net5.0-windows10.0.17763.0\Microsoft.WinUI\Themes\generic.xaml` which is based inside the `WindowChromeStyle` style. I then use Win32 API to wire up the buttons.

This whole view can _almost_ be replaced with a single line of 
```
<ContentControl x:Name="MainContentControl" Style="{StaticResource WindowChromeStyle}" />
```
however, by doing this I can't wire up the click events inside this ContentControl. Everything used inside `WindowChromeStyle` is pretty much copied (although disabled) [here](https://github.com/beeradmoore/WindowsAppSDK_NavColorTest/blob/main/NavTest/NavTest/UserControls/FakeMinMaxCloseButtons.xaml#L14-L196).

As you can see inside it there is 3 buttons `MinimizeButton`, `MaximizeButton` and `CloseButton`. I had some [failed attempts](https://github.com/beeradmoore/WindowsAppSDK_NavColorTest/blob/main/NavTest/NavTest/UserControls/FakeMinMaxCloseButtons.xaml.cs#L50-L61) of trying to reference buttons and styles but nothing seemed to work.

Ideally I'd like to reference the style inside WindowsAppSDK rather than copy it (which is what the current implementation is) so if there is any updates in the future I will have those updates.

So the TL;DR; question is, given the `ContentControl` above which has a style applied to it of `WindowChromeStyle`, how do you reference the buttons that get defined inside it? Is that possible? Is it possible in a future update to add commands or something else to these buttons so they could be accessed externally?

**Comment by BenJKuhn** [comment, marked as answer]
I've moved this over to the WinUI repo, someone should be able to help answer your question over there:

https://github.com/microsoft/microsoft-ui-xaml/issues/6738

**Reply by beeradmoore** [reply]
Ah cheers. I posted it here because this was the WinUI3 repo at one point and that's where UI title bar discussions were. I'll subscribe to the new topic and keep investigating here.
